src/main/java/org/teamresistance/frc/Gyro3D.java

```diff
@@ -0,0 +1,42 @@
+package org.teamresistance.frc;
+
+import com.kauailabs.navx.frc.AHRS;
+import edu.wpi.first.wpilibj.SPI;
+import org.strongback.components.AngleSensor;
+
+/**
+ * @author Shreya Ravi
+ */
+public class Gyro3D extends AHRS implements AngleSensor {
+
+  public Gyro3D(SPI.Port port) {
+    super(port);
+  }
+
+
+  @Override
+  public double getAngle() {
+    return getYawVal();
+  }
+
+  private double normalize(double angle) {
+    return ((angle % 360) + 360) % 360;
+  }
+
+  public double getYawVal() {
+    return normalize((double) super.getYaw());
+  }
+
+  public double getRollVal() {
+    return normalize((double) super.getRoll());
+  }
+
+  public double getPitchVal() {
+    return normalize((double) super.getPitch());
+  }
+
+  public boolean isAngle(double threshold, double angle) {
+    double currAngle = getYawVal();
+    return ((Math.abs(currAngle - angle) <= threshold) ? true : false);
```

Comment:
I'm silly, I just realized you can directly return the result of the boolean check. Both the ternary and if/else are redundant, since you can just do this:

```java
return Math.abs(currAngle - angle) <= threshold;
```

I would also take it a step further, and inline `currAngle`, eliminating the need for the intermediate variable since the statement appears to be easy enough to understand. I went ahead and used `getAngle` because of the comment earlier about `getYawVal` not being necessary.

```java
return Math.abs(getAngle() - angle) <= threshold;
```